src/games/supported/Galaxian.cpp

```diff
@@ -122,3 +122,33 @@ void GalaxianSettings::loadState(Deserializer & ser) {
   m_lives = ser.getInt();
 }
 
+// returns a list of mode that the game can be played in
+ModeVect GalaxianSettings::getAvailableModes() {
+    ModeVect modes = { 1, 2, 3, 4, 5, 6, 7, 8, 9 };
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void GalaxianSettings::setMode(game_mode_t mode, System &system,
+                              std::unique_ptr<StellaEnvironmentWrapper> environment) {
+
+    if (mode >= 1 && mode <= 9) {
+        // press select until the correct mode is reached
+        while (mode != static_cast<unsigned>(readRam(&system, 0xB3))) {
+            environment->pressSelect();
+        }
+        //reset the environment to apply changes.
+        environment->softReset();
+    } 
+    // ignore 0 since that's the default even though it's not a 
+    // valid mode
+    else if (mode != 0) 
```

Comment:
The default mode is `0`, here you are forcing the user to set the mode, aren't you? We don't want that, we want the default mode to be used when `setMode` and `setDifficulty` is not called.

Reply:
I've changed it so that if mode 0 is passed in it sets it to mode 1.

I think the patch in pull #236 is a better way of handling this though. Mode 0 doesn't really exist.